Theme colors

KindaOK opened the pull request:
Add easy theming to the site. All colors can be configured from the `theme.js` file.

KindaOK committed:
Create theme file and add colors

Color names need something better so it's easier to integrate them.

KindaOK committed:
Create a mostly-populated light theme

KindaOK committed:
Add theme provider to app

KindaOK committed:
Create wrappers for theme management

Themes are stored in styled component themes with a context for
changing the active theme.

Also improve theme styles and create dark theme for testing.

KindaOK committed:
Specify default theme selection process

KindaOK committed:
Convert About to a themed page

Still missing theming for the svgs.

KindaOK committed:
Apply theming to the about page

Currently an issue with styled components bugging with having nth-of-type
combined with the devices template. Will probably create a manually
mirrored version instead to fix it.

KindaOK committed:
Split Mission Statement into two elements

Mirroring is now manually applied to get around the weird styled
components bug.

KindaOK committed:
Convert Home into a themeable page

KindaOK committed:
Remove old Home class component

KindaOK committed:
Themify leadership page

Also remove an old import and simplify card container

KindaOK committed:
Add theming to Teams Page

KindaOK committed:
Add lead portrait border to theme

KindaOK committed:
Refactor SubTheme to use a theme function instead of object

A theme object will cause styled-components to copy down props from
the outer theme into the child theme. This pollutes the namespace and
allows naughty programmers to reach too far inside the abstraction.

A theme function will not copy props from the outer theme, meaning
that we can limit what props the programmer can access.

KindaOK committed:
Remove duplicate SubTheme

"leads" was nested in Leadership and in CardContainer

KindaOK committed:
Add theming to the Navbar

Also figured out masking for the logo, so it's easy to adjust the
logo color.

KindaOK committed:
Add theming to FAQ

Also figured out part of the SVG issue we were having with about

KindaOK committed:
Fix bug with about styles and svgs

We split the MissionStatementElement into a mirrored version because
nested styles weren't working. It had to do with the svg encoding.

KindaOK committed:
Add initial theming to Contact

Still need to rewrite svgs to accept themes and the text to be in html

KindaOK committed:
Add Theming to Contact

SVGs were simplified (remade if necessary) and now have colors
templated in for convenience of adjustment. Old ones were removed
since they are not super useful now.

kgurushankar commented:
At the moment, you have a number of files where the line width appears to be 80, and others where it is unbounded. Fix for consistency please :) 

If you have the time, would also appreciate some kind of rc file for the formatter + a script to invoke it in place (to potentially be used as a git hook/CI check)

KindaOK commented:
Formatting script is `prettier --write .`. There's a bunch of [existing hooks](https://prettier.io/docs/en/precommit.html) that use it.

Yeah prettier doesn't reformat strings automatically because that would change the AST. Fixing some of the string in JSX makes sense though.

KindaOK committed:
Add theming to PageNotFound (mostly)

SubpageOuter still doesn't have theming and it's probably going to
be a pain.

KindaOK committed:
Update theme for FAQ for heading

KindaOK committed:
Add basic theming to Events

Not sure if it's worth the time to theme the rest of the calendar if
it's subject to change soon.

KindaOK committed:
Move ThemeComponents to src/components

KindaOK committed:
Removed unused import

KindaOK committed:
Copy new version of light theme as a base

KindaOK committed:
Change theme to sync to system by default

KindaOK committed:
Adjust theming of several components

Fix some issues of colors still being hardcoded and/or not properly
applied.

KindaOK committed:
Set body background color from theme

KindaOK committed:
Add theming to home geometry

KindaOK committed:
Add theming to mailing list

KindaOK committed:
Add a theme switcher

Also made the theme switching look really nice except for the svgs.
Not sure how to best fix that :(

KindaOK committed:
Removed unused import

KindaOK committed:
Merge branch 'dev' into theme-colors

KindaOK committed:
Fix minor style issues in teams

Layering and theming.
